components/features/MentorRanking/MentorRanking.js

```diff
@@ -18,6 +20,18 @@ export default function Leaderboard() {
     setAnchorEl(null);
   };
 
+  useEffect(() => {
+    fetchTopEndorsements();
+  }, [])
+
+  const fetchTopEndorsements = () => {
+    getTopEndorsements().then(res => {
+      setTopEndorsements(res)
+    }).catch(err => {
+      console.log('Error getting data')
+    })
+  }
```

Comment:
If we are only calling getTopEndorsements once then we can put this in the useEffect instead of wrapping it in a function.